lua/wire/von.lua

```diff
@@ -354,7 +354,14 @@ _deserialize = {
 		a = find(s, "[;:}~]", i)
 
 		if a then
-			return tonumber(sub(s, i, a - 1)) or error("vON: Number definition does not contain a valid number!"), a - 1
+			a = a - 1
+			local c = sub(s, i, a)
+
+			if c == "inf" then return math.huge, a
+			elseif c == "-inf" then return -math.huge, a
+			elseif c == "nan" then return 0/0, a end
+
+			return tonumber(c) or error("vON: Number definition does not contain a valid number!"), a
```

Comment:
Should store tonumber(c) in a local and check if it's nil first before checking those three strings, since valid numbers will be the most common case.

Reply:
I also want to test of gmod really doesn't tonumber inf/nan correctly. Should open a garrysmod issue perhaps.

Reply:
Would you be against it being a chain of conditional operators?